src/components/org/PositionTimeline/model.ts

```diff
@@ -0,0 +1,59 @@
+export type TimelineSplit = {
+    id: string;
+    externalId: string;
+    workload: number;
+    appliesFrom: Date;
+    appliesTo: Date;
+    parentPositionId: string | null;
+    taskOwnerIds: string[] | null;
+    reportsToIds?: string[] | undefined;
+    obs: string;
+    rotationId: string | null;
+    properties: { workPackId?: string; hasRequest?: boolean } | null;
+    type: 'Normal' | 'Rotation';
+    location: {
+        id: string;
+    } |  null;
+    assignedPerson: {
+        azureUniqueId?: string;
+        mail?: string | null;
+        name?: string;
+    } | null;
+    positionId: string;
+    calendar?: string | null;
+};
+
+export type TimelinePosition = {
+    id?: string;
+    name: string;
+    externalId: string | null;
+    properties: {
+        isSupport?: boolean;
+        hideInTree?: boolean;
+        resourceType?: 'normal' | 'jointVenture' | 'enterprise';
+    };
+    basePosition: {
+        id: string;
+        department?: string;
+    };
+    instances: TimelineSplit[];
+    isTaskOwner: boolean;
+};
+
+export type SplitDetails = any & { appliesFrom: Date; appliesTo: Date };
+
+export type RotationGroups = Record<string, TimelineSplit[]>;
+
+export type PersonSlotProps<T> = {
+    item: T;
+};
+
+export type InfoSlotProps<T> = {
+    item: T;
+};
+
+export type ActionSlotProps<T> = {
+    value: T;
+};
```

Comment:
Could this be one type?

Reply:
Does this need to be generic? Could T be TimelineSplit in all cases?

Reply:
Replaced by a common prop type with a split and isSelected prop.